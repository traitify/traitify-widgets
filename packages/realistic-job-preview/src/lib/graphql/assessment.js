export const create = `
  mutation createAssessment(
    $localeKey: String!,
    $profileID: String!,
    $surveyID: String!
  ) {
    createAssessment(
      localeKey: $localeKey,
      profileId: $profileID,
      surveyId: $surveyID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const get = `
  query assessment(
    $assessmentID: ID!
  ) {
    assessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const path = "/realistic-job-previews/graphql";

export const start = `
  mutation startAssessment(
    $assessmentID: ID!
  ) {
    startAssessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const update = `
  mutation updateAssessmentAnswer(
    $answers: [AssessmentAnswer],
    $assessmentID: String!
  ) {
    updateAssessmentAnswer(
      answers: $answers,
      assessmentId: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;
