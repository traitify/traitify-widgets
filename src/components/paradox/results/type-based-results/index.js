import PropTypes from "prop-types";
import PersonalityBase from "components/results/type-based-results/personality-base";
import PersonalityDetails from "components/results/type-based-results/personality-details";
import PersonalityTraits from "components/results/type-based-results/personality-traits";
import PersonalityTypes from "components/results/type-based-results/personality-types";
import withTraitify from "lib/with-traitify";
import style from "../style.scss";

function TypeBasedResults({setElement, ...props}) {
  return (
    <section className={style.container} ref={setElement}>
      <PersonalityBase {...props} />
      <PersonalityTypes {...props} />
      <PersonalityTraits {...props} />
      <PersonalityDetails {...props} />
    </section>
  );
}

TypeBasedResults.propTypes = {
  setElement: PropTypes.func.isRequired
};

export {TypeBasedResults as Component};
export default withTraitify(TypeBasedResults);
