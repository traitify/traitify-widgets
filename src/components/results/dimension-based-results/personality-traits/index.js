import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import PersonalityTrait from "../personality-trait";
import style from "./style.scss";

class PersonalityTraits extends Component {
  static defaultProps = {assessment: null};
  static propTypes = {
    assessment: PropTypes.shape({
      personality_traits: PropTypes.arrayOf(
        PropTypes.shape({
          personality_trait: PropTypes.shape({
            id: PropTypes.string.isRequired
          }).isRequired
        }).isRequired
      )
    }),
    getOption: PropTypes.func.isRequired,
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  };
  componentDidMount() {
    this.props.ui.trigger("PersonalityTraits.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityTraits.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }

    const perspective = this.props.getOption("perspective") || "firstPerson";
    const thirdPersonCheck = perspective === "thirdPerson";
    const traits = this.props.assessment.personality_traits;

    return (
      <div className={style.traits}>
        <h4 className={style.title}>{this.props.translate("most_represented_traits")}</h4>
        {thirdPersonCheck && (
          <div className={style.traitsDefinition}>{this.props.translate("most_represented_traits_definition")}</div>
        )}
        {traits.slice(0, 5).map((trait) => (
          <PersonalityTrait key={trait.personality_trait.id} trait={trait} {...this.props} />
        ))}
        <h4 className={style.title}>{this.props.translate("least_represented_traits")}</h4>
        {thirdPersonCheck && (
          <div className={style.traitsDefinition}>{this.props.translate("least_represented_traits_definition")}</div>
        )}
        {traits.slice(-5).map((trait) => (
          <PersonalityTrait key={trait.personality_trait.id} trait={trait} {...this.props} />
        ))}
      </div>
    );
  }
}

export {PersonalityTraits as Component};
export default withTraitify(PersonalityTraits);
