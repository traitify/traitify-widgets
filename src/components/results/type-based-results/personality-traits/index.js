import PropTypes from "prop-types";
import {Component} from "react";
import {Component as Paradox} from "components/paradox/results/personality/trait/list";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import PersonalityTrait from "../personality-trait";
import style from "./style.scss";

class PersonalityTraits extends Component {
  static defaultProps = {assessment: null};
  static propTypes = {
    assessment: PropTypes.shape({
      personality_traits: PropTypes.arrayOf(
        PropTypes.shape({
          personality_trait: PropTypes.shape({
            id: PropTypes.string.isRequired
          }).isRequired
        })
      )
    }),
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  };
  constructor(props) {
    super(props);

    this.state = {showMore: false};
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityTraits.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityTraits.updated", this);
  }
  onClick = () => {
    this.setState((state) => ({showMore: !state.showMore}), () => {
      const key = this.state.showMore ? "showMore" : "showLess";

      this.props.ui.trigger(`PersonalityTraits.${key}`, this);
    });
  };
  render() {
    if(!this.props.isReady("results")) { return null; }

    const text = this.props.translate(this.state.showMore ? "show_less" : "show_more");
    let traits = this.props.assessment.personality_traits;

    if(!this.state.showMore) { traits = traits.slice(0, 8); }

    return (
      <div className={style.traits}>
        {traits.map((trait) => (
          <PersonalityTrait key={trait.personality_trait.id} trait={trait} {...this.props} />
        ))}
        <p className={style.center}>
          <button className={style.toggle} onClick={this.onClick} type="button">{text}</button>
        </p>
      </div>
    );
  }
}

export {PersonalityTraits as Component};
export default withTraitify(PersonalityTraits, {paradox: Paradox});
