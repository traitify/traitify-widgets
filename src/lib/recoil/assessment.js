import {selector, selectorFamily} from "recoil";
import {assessmentsState} from "./assessments";
import {
  activeState,
  activeIDState,
  activeTypeState,
  cacheState,
  graphqlState,
  httpState,
  localeState,
  safeCacheKeyState
} from "./base";

// TODO: Return error instead of null for cognitive and external
// return {errors: response.errors, id};
export const cognitiveAssessmentQuery = selectorFamily({
  get: (id) => async({get}) => {
    if(!id) { return null; }

    const cache = get(cacheState);
    const cacheKey = get(safeCacheKeyState({id, type: "assessment"}));
    const cached = cache.get(cacheKey);
    if(cached) { return cached; }

    const GraphQL = get(graphqlState);
    const http = get(httpState);
    const params = {
      query: GraphQL.cognitive.get,
      variables: {localeKey: get(localeState), testID: id}
    };
    const response = await http.post({path: GraphQL.cognitive.path, params});
    if(response.errors) {
      console.warn("cognitive-assessment", response.errors); /* eslint-disable-line no-console */
      return null;
    }

    const assessment = response.data.cognitiveTest;
    if(!assessment?.completed) { return assessment; }

    cache.set(cacheKey, assessment);

    return assessment;
  },
  key: "assessment/cognitive"
});

export const externalAssessmentQuery = selectorFamily({
  get: (id) => async({get}) => {
    if(!id) { return null; }

    const cache = get(cacheState);
    const cacheKey = get(safeCacheKeyState({id, type: "assessment"}));
    const cached = cache.get(cacheKey);
    if(cached) { return cached; }

    const GraphQL = get(graphqlState);
    const http = get(httpState);
    const query = {
      params: {
        query: GraphQL.external.get,
        variables: {id}
      },
      path: GraphQL.external.path
    };
    if(http.version === "v1") { query.version = GraphQL.external.version; }

    const response = await http.post(query);
    if(response.errors) {
      console.warn("external-assessment", response.errors); /* eslint-disable-line no-console */
      return null;
    }

    const assessment = response.data.getAssessment;
    if(!assessment?.completedAt) { return assessment; }

    cache.set(cacheKey, assessment);

    return assessment;
  },
  key: "assessment/external"
});

export const personalityAssessmentQuery = selectorFamily({
  get: (id) => async({get}) => {
    if(!id) { return null; }

    const cache = get(cacheState);
    const cacheKey = get(safeCacheKeyState({id, type: "assessment"}));
    const cached = cache.get(cacheKey);
    if(cached) { return cached; }

    const params = {
      data: "archetype,blend,instructions,recommendation,slides,types,traits",
      locale_key: get(localeState)
    };
    const http = get(httpState);
    const response = await http.get({path: `/assessments/${id}`, params});
    if(!response?.completed_at) { return response; }

    cache.set(cacheKey, response);

    return response;
  },
  key: "assessment/personality"
});

export const genericAssessmentQuery = selectorFamily({
  get: (id) => async({get}) => {
    if(!id) { return null; }

    const cache = get(cacheState);
    const cacheKey = get(safeCacheKeyState({id, type: "assessment"}));
    const cached = cache.get(cacheKey);
    if(cached && !cached.completed) { return cached; }

    const GraphQL = get(graphqlState);
    const http = get(httpState);
    const params = {
      query: GraphQL.generic.questions,
      variables: {assessmentID: id}
    };

    const response = await http.post({path: GraphQL.generic.path, params});
    if(response.errors) {
      console.warn("generic-assessment", response.errors); /* eslint-disable-line no-console */
      return null;
    }

    const assessment = response.data.genericSurveyQuestions;
    if(!assessment?.completedAt) { return assessment; }

    const query = GraphQL.generic.result;
    const variables = {assessmentID: id};

    const resultResponse = await http.post(GraphQL.generic.path, {query, variables});
    if(resultResponse.errors) { return null; }
    const result = resultResponse.data.genericAssessmentResult.assessment;

    cache.set(cacheKey, result);

    return result;
  },
  key: "assessment/generic"
});

export const assessmentQuery = selectorFamily({
  get: ({id, surveyType}) => async({get}) => {
    if(surveyType === "cognitive") { return get(cognitiveAssessmentQuery(id)); }
    if(surveyType === "external") { return get(externalAssessmentQuery(id)); }
    if(surveyType === "generic") { return get(genericAssessmentQuery(id)); }
    if(surveyType === "personality") { return get(personalityAssessmentQuery(id)); }

    return null;
  },
  key: "assessment"
});

export const activeAssessmentQuery = selector({
  get: async({get}) => {
    const id = get(activeIDState);
    const surveyType = get(activeTypeState);

    return get(assessmentQuery({id, surveyType}));
  },
  key: "assessment/active"
});

export const completedAssessmentQuery = selectorFamily({
  get: ({surveyType}) => async({get}) => {
    if(!surveyType) {
      const active = get(activeState);
      if(!active?.completed) { return null; }

      return get(activeAssessmentQuery);
    }

    const assessments = get(assessmentsState);
    if(!assessments) { return null; }

    const assessment = assessments
      .find(({completed, surveyType: type}) => type === surveyType && completed);
    if(!assessment) { return null; }

    return get(assessmentQuery({id: assessment.id, surveyType}));
  },
  key: "assessment/completed"
});
