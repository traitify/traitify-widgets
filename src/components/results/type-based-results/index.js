import {
  faThLarge,
  faUser
} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import {Component as Paradox} from "components/paradox/results/type-based-results";
import Icon from "lib/helpers/icon";
import withTraitify from "lib/with-traitify";
import PersonalityBase from "./personality-base";
import PersonalityDetails from "./personality-details";
import PersonalityTraits from "./personality-traits";
import PersonalityTypes from "./personality-types";
import style from "../style.scss";

function TypeBasedResults(props) {
  const {getOption, translate} = props;
  const allowHeaders = getOption("allowHeaders");

  return (
    <section>
      {allowHeaders && (
        <p className={style.heading}><Icon icon={faUser} /> {translate("personality_blend")}</p>
      )}
      <PersonalityBase {...props} />
      {allowHeaders && (
        <p className={style.heading}><Icon icon={faThLarge} /> {translate("personality_breakdown")}</p>
      )}
      <PersonalityTypes {...props} />
      {allowHeaders && (
        <p className={style.heading}><Icon icon={faUser} /> {translate("personality_traits")}</p>
      )}
      <PersonalityTraits {...props} />
      {allowHeaders && (
        <p className={style.heading}><Icon icon={faThLarge} /> {translate("personality_details")}</p>
      )}
      <PersonalityDetails {...props} />
    </section>
  );
}

TypeBasedResults.propTypes = {
  getOption: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired
};

export default withTraitify(TypeBasedResults, {paradox: Paradox});
