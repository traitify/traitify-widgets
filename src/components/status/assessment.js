import {faCheck} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import {useSetRecoilState} from "recoil";
import Icon from "components/common/icon";
import get from "lib/common/object/get";
import useListener from "lib/hooks/use-listener";
import useOption from "lib/hooks/use-option";
import {activeState} from "lib/recoil";
import style from "./style.scss";

// TODO: Extract text to translate
const translations = {
  complete: "Complete",
  loading: "Loading", // Already translated
  status: {
    start: "Start Assessment"
  },
  survey: {
    cognitive_assessment: "Cognitive Assessment",
    external_assessment: "External Assessment",
    personality_assessment: "Personality Assessment"
  }
};

function Button({assessment}) {
  const listener = useListener();
  const options = useOption("status") || {};
  const redirect = get(options, "allowRedirect", true);
  const setActive = useSetRecoilState(activeState);

  if(assessment.completed) {
    return <button disabled={true} type="button">{translations.complete}</button>;
  }

  if(assessment.skipped) {
    return <button disabled={true} type="button">Skipped</button>;
  }

  if(assessment.link && redirect) {
    return <a href={assessment.link}>{translations.status.start}</a>;
  }

  if(assessment.loading) {
    return <button disabled={true} type="button">{translations.loading}</button>;
  }

  const start = () => {
    listener.trigger("Survey.start", {assessment});
    setActive({...assessment});
  };

  return <button onClick={start} type="button">{translations.status.start}</button>;
}

Button.propTypes = {
  assessment: PropTypes.shape({
    completed: PropTypes.bool,
    link: PropTypes.string,
    loading: PropTypes.bool,
    skipped: PropTypes.bool
  }).isRequired
};

function Assessment({assessment}) {
  const surveyName = assessment.surveyName || translations.survey[`${assessment.surveyType}_assessment`];

  return (
    <div className={[style.assessment, assessment.completed && style.inactive].filter(Boolean).join(" ")}>
      <div>
        {assessment.completed && <Icon className={style.icon} icon={faCheck} />}
        <span className={style.text}>{surveyName}</span>
      </div>
      <Button assessment={assessment} />
    </div>
  );
}

Assessment.propTypes = {
  assessment: PropTypes.shape({
    completed: PropTypes.bool,
    skipped: PropTypes.bool,
    surveyName: PropTypes.string,
    surveyType: PropTypes.string.isRequired
  }).isRequired
};

export default Assessment;
