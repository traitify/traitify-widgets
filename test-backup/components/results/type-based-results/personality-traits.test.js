import {Component} from "components/results/type-based-results/personality-traits";
import ComponentHandler from "support/component-handler";
import assessment from "support/json/assessment/type-based.json";

jest.mock("components/results/type-based-results/personality-trait", () => ((props) => (
  <div className="mock">Trait - {props.trait.personality_trait.name}</div>
)));
jest.mock("lib/with-traitify", () => ((value) => value));

describe("PersonalityTraits", () => {
  let props;

  beforeEach(() => {
    props = {
      assessment,
      isReady: jest.fn().mockName("isReady").mockImplementation(() => true),
      translate: jest.fn().mockName("translate").mockImplementation((value) => value),
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      const component = new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityTraits.initialized", component.instance);
    });

    it("triggers update", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.updateProps();

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityTraits.updated", component.instance);
    });

    it("triggers show less", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.updateState({showMore: true});
      component.instance.onClick();

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityTraits.showLess", component.instance);
    });

    it("triggers show more", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.instance.onClick();

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityTraits.showMore", component.instance);
    });
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if not ready", () => {
    props.assessment = null;
    props.isReady.mockImplementation(() => false);
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders more traits", () => {
    const component = new ComponentHandler(<Component {...props} />);
    component.updateState({showMore: true});

    expect(component.tree).toMatchSnapshot();
  });
});
