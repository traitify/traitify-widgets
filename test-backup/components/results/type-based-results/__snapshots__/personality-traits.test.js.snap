// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalityTraits renders component 1`] = `
<div
  className="traits"
>
  <div
    className="mock"
  >
    Trait - 
    Alert
  </div>
  <div
    className="mock"
  >
    Trait - 
    Organic
  </div>
  <div
    className="mock"
  >
    Trait - 
    Loyal
  </div>
  <div
    className="mock"
  >
    Trait - 
    Relaxed
  </div>
  <div
    className="mock"
  >
    Trait - 
    Serious
  </div>
  <div
    className="mock"
  >
    Trait - 
    Altruistic
  </div>
  <div
    className="mock"
  >
    Trait - 
    Intellectually-Driven
  </div>
  <div
    className="mock"
  >
    Trait - 
    Steady
  </div>
  <p
    className="center"
  >
    <button
      className="toggle"
      onClick={[Function]}
      type="button"
    >
      show_more
    </button>
  </p>
</div>
`;

exports[`PersonalityTraits renders more traits 1`] = `
<div
  className="traits"
>
  <div
    className="mock"
  >
    Trait - 
    Alert
  </div>
  <div
    className="mock"
  >
    Trait - 
    Organic
  </div>
  <div
    className="mock"
  >
    Trait - 
    Loyal
  </div>
  <div
    className="mock"
  >
    Trait - 
    Relaxed
  </div>
  <div
    className="mock"
  >
    Trait - 
    Serious
  </div>
  <div
    className="mock"
  >
    Trait - 
    Altruistic
  </div>
  <div
    className="mock"
  >
    Trait - 
    Intellectually-Driven
  </div>
  <div
    className="mock"
  >
    Trait - 
    Steady
  </div>
  <div
    className="mock"
  >
    Trait - 
    Protective
  </div>
  <div
    className="mock"
  >
    Trait - 
    Inquisitive
  </div>
  <div
    className="mock"
  >
    Trait - 
    Earthy
  </div>
  <div
    className="mock"
  >
    Trait - 
    Spontaneous
  </div>
  <div
    className="mock"
  >
    Trait - 
    Observant
  </div>
  <div
    className="mock"
  >
    Trait - 
    Unconventional
  </div>
  <div
    className="mock"
  >
    Trait - 
    Optimistic
  </div>
  <div
    className="mock"
  >
    Trait - 
    Independent
  </div>
  <div
    className="mock"
  >
    Trait - 
    Leader
  </div>
  <div
    className="mock"
  >
    Trait - 
    Future-Focused
  </div>
  <div
    className="mock"
  >
    Trait - 
    Visionary
  </div>
  <div
    className="mock"
  >
    Trait - 
    Analytical
  </div>
  <div
    className="mock"
  >
    Trait - 
    Deliberate
  </div>
  <div
    className="mock"
  >
    Trait - 
    Likes Novelty
  </div>
  <div
    className="mock"
  >
    Trait - 
    Deep
  </div>
  <div
    className="mock"
  >
    Trait - 
    Open-Minded
  </div>
  <div
    className="mock"
  >
    Trait - 
    Prefers Working Alone
  </div>
  <div
    className="mock"
  >
    Trait - 
    Physically-Driven
  </div>
  <div
    className="mock"
  >
    Trait - 
    High-Energy
  </div>
  <div
    className="mock"
  >
    Trait - 
    Daring
  </div>
  <div
    className="mock"
  >
    Trait - 
    Conventional
  </div>
  <div
    className="mock"
  >
    Trait - 
    Meticulous
  </div>
  <div
    className="mock"
  >
    Trait - 
    Compulsive
  </div>
  <div
    className="mock"
  >
    Trait - 
    Creative
  </div>
  <div
    className="mock"
  >
    Trait - 
    Confident
  </div>
  <div
    className="mock"
  >
    Trait - 
    Goal-Oriented
  </div>
  <div
    className="mock"
  >
    Trait - 
    Flexible
  </div>
  <div
    className="mock"
  >
    Trait - 
    Detail-Oriented
  </div>
  <div
    className="mock"
  >
    Trait - 
    Patient
  </div>
  <div
    className="mock"
  >
    Trait - 
    Process-Oriented
  </div>
  <div
    className="mock"
  >
    Trait - 
    Concrete
  </div>
  <div
    className="mock"
  >
    Trait - 
    Likes Familiarity
  </div>
  <div
    className="mock"
  >
    Trait - 
    Aggressive
  </div>
  <div
    className="mock"
  >
    Trait - 
    Practical
  </div>
  <div
    className="mock"
  >
    Trait - 
    Whimsical
  </div>
  <div
    className="mock"
  >
    Trait - 
    Realistic
  </div>
  <div
    className="mock"
  >
    Trait - 
    Particular
  </div>
  <div
    className="mock"
  >
    Trait - 
    Direct
  </div>
  <div
    className="mock"
  >
    Trait - 
    Passionate
  </div>
  <div
    className="mock"
  >
    Trait - 
    Competitive
  </div>
  <div
    className="mock"
  >
    Trait - 
    Outgoing
  </div>
  <div
    className="mock"
  >
    Trait - 
    Prefers Working with Others
  </div>
  <div
    className="mock"
  >
    Trait - 
    Intuitive
  </div>
  <div
    className="mock"
  >
    Trait - 
    Stable
  </div>
  <div
    className="mock"
  >
    Trait - 
    Empathic
  </div>
  <div
    className="mock"
  >
    Trait - 
    Rational
  </div>
  <div
    className="mock"
  >
    Trait - 
    Sensitive
  </div>
  <div
    className="mock"
  >
    Trait - 
    Decisive
  </div>
  <p
    className="center"
  >
    <button
      className="toggle"
      onClick={[Function]}
      type="button"
    >
      show_less
    </button>
  </p>
</div>
`;

exports[`PersonalityTraits renders nothing if not ready 1`] = `null`;
